src/domain/profile.ts

```diff
@@ -45,13 +46,41 @@ export class Profile {
     this.role = role;
   }
 
+  static async create(
+    name: string,
+    email: string,
+    password: string,
+    hasSecondShot: boolean,
+    document: IDocument,
+    phone = '',
+    address = '',
+    dayOfSecondShot: Date | undefined = undefined,
+    role: Role = new Role(RoleType.User)
+  ): Promise<Profile> {
+    return new Profile(
+      name,
+      email,
+      await Profile.hashPassword(password),
+      hasSecondShot,
+      document,
+      phone,
+      address,
+      dayOfSecondShot,
+      role
+    );
+  }
+
```

Comment:
Aqui não é exatamente um pedido de mudança, eu só to estranhando ter que criar um segundo método para instanciar, por causa da replicação de código. Imagino que isso seja porque esse método teve que ser async, e o constructor não pode ser async. Cê pode me explicar o motivo disso?

Reply:
Não é só por ser async (mas também por isso, não queria usar a versão síncrona do bcryptjs. Mas é mais por separação de responsabilidades, porque o construtor tá sendo utilizado como um método de desserialização, e esse `create` como uma fábrica.
O construtor é pra ser usado a partir de um conjunto de dados que já representa um perfil (já possui id, senha em hash e etc), o `create` é pra ser usado a partir de um conjunto de dados que se pretende virar um perfil (não tem id, e a senha ainda está em plain-text).